### ISSUE TITLE
A new ESM loader equivalent of 'ts-node/register/transpile-only'

### ISSUE THREAD

**Issue by concision** [TypeStrong/ts-node#1101]
@cspotcode 

Continuing the discussion from [#1007](https://github.com/TypeStrong/ts-node/issues/1007#issuecomment-671155266).

<br>

CommonJS has the file shortcut `ts-node/register/transpile-only` for installing the TypeScript loader without type checking. An ESM loader which has the equivalent functionality should be created and exposed to the end user. 

https://github.com/TypeStrong/ts-node/blob/ce7c323557c15bf878bbc4dd5b2c0aac219fca4b/esm.mjs#L7

The code from [esm.mjs](https://github.com/TypeStrong/ts-node/blob/master/esm.mjs#L7) can be duplicated in a new file, but with the `transpileOnly` flag set:

```ts
export const {resolve, getFormat, transformSource} = esm.registerAndCreateEsmHooks({transpileOnly: true})
```

<br>

What should the loader filename look like? There is `ts-node/esm.mjs`, should it be `ts-node/esm/transpile-only.mjs`?

**Comment by cspotcode** [comment]
`ts-node/esm/transpile-only.mjs` makes sense to me.  A user does `node --loader ts-node/esm/transpile-only ./foo.ts`

**Comment by concision** [comment]
Resolved with #1102
